Automattic/woocommerce-payments
# fix: save platform checkout info on blocks

Fixes https://github.com/Automattic/woocommerce-payments/issues/7203

#### Changes proposed in this Pull Request

Removing some logic related to checkout blocks that doesn't seem to be relevant anymore, and that instead is creating a bug where the billing phone number is wiped out when WooPay's "save my info" box is checked on the blocks checkout.









#### Testing instructions





- Ensure you are using one of the UPE variations (Deferred Intent UPE is good to test with, since it'll become the default)
- Add some product(s) to your cart
- Go to the blocks checkout page
- Fill in the checkout fields, but don't enter the phone number in the "main" checkout form
- Click "Save my info" to save the info on WooPay
- Enter the phone number for WooPay (it shouldn't be filled in, since you didn't enter it in the previous step)
- Place the order
- Ensure that the customer is created on WooPay with the correct phone number (you'll need the webhooks working for this), and that the order is created in the backend without the phone number
- Log out, now attempt to do the same steps with a different email address, but fill in the phone number in the "main" checkout form
- Click "Save my info" to save the info on WooPay, but, just for fun, use a different phone number
- Place the order
- Ensure that the customer is created on WooPay with the correct phone number, and that the order is created in the backend with the correct phone number

-------------------


**Post merge**
